### PR TITLE
remove liquidity imbalance agent created from template

### PR DESCRIPTION
**Pull request by yasmine9**
Create remove liquidity imbalance agent using the  agent template. 
- Agent refactored using `GAM`. 
- Finding metadata improved. 
- More tests cases added.

### CODE REVIEW

File: Curve-finance-suite/Remove-Imbalace-Liquidity/package.json [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    "push": "forta-agent push",
+    "disable": "forta-agent disable",
+    "enable": "forta-agent enable",
+    "keyfile": "forta-agent keyfile",
+    "test": "jest"
+  },
+  "dependencies": {
+    "bignumber.js": "^9.0.1",
+    "forta-agent": "^0.0.32"
+  },
```
Comment (e1Ru1o) [inline Curve-finance-suite/Remove-Imbalace-Liquidity/package.json:24]: `forta-agent-tools` dependency is missing

File: Curve-finance-suite/Remove-Imbalace-Liquidity/package.json [hunk 2]
```diff
@@ -0,0 +1,33 @@
+{
+  "name": "forta-agent-starter",
+  "version": "0.0.1",
+  "description": "Forta Agent Typescript starter project",
```
Comment (e1Ru1o) [inline Curve-finance-suite/Remove-Imbalace-Liquidity/package.json:4]: `name` needs to be changed to the agent id, you can choose one. We usually use `<protocol>-<agent#>` so `curve-2` will be fine.
Also add a small description of the agent in `description`

File: Curve-finance-suite/Remove-Imbalace-Liquidity/src/agent.spec.ts [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+describe('Remove liquidity imbalance Agent tests suite', () => {
+  const handleTransaction: HandleTransaction = provideRemoveLiquidityImbalanceAgent(ALERTID, ADDRESS);;
+  const data = encodeParameters(
+    ["uint256[3]","uint256[3]","uint256", "uint256"],
+    [ [1,2,3],[1,2,3],3,5]
+  );
+
+  beforeAll(() => {
+
+  });
```
Comment (e1Ru1o) [inline Curve-finance-suite/Remove-Imbalace-Liquidity/src/agent.spec.ts:38]: If this will remains empty you can remove it

**Review by e1Ru1o — CHANGES_REQUESTED** [review]

File: Curve-finance-suite/Remove-Imbalace-Liquidity/src/agent.ts [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+  address: string
+): HandleTransaction => provideEventCheckerHandler(
+    createFindingGenerator(alertID),
+    iface.getEvent('RemoveLiquidityImbalance').format('sighash'),
+    address
+  );
+
+export default {
+  handleTransaction: provideRemoveLiquidityImbalanceAgent(
+    "CURVE-2",
```
Comment (mauricio1802) [inline Curve-finance-suite/Remove-Imbalace-Liquidity/src/agent.ts:50]: Can we change this to use the numbers from the table at `https://www.notion.so/nethermind/CurveDAO-7807d9f852ac42f2973c5aa4747a80af`?

**Review by e1Ru1o — APPROVED** [review]

File: Curve-finance-suite/Remove-Imbalace-Liquidity/src/agent.ts [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+  FindingSeverity, 
+  FindingType 
+} from 'forta-agent'
+import {
+  decodeParameters,
+  FindingGenerator,
+  provideEventCheckerHandler,
+} from "forta-agent-tools";
+
+export const CONTRACT_ADDRESS = '0xbEbc44782C7dB0a1A60Cb6fe97d0b483032FF1C7';
```
Comment (mauricio1802) [inline Curve-finance-suite/Remove-Imbalace-Liquidity/src/agent.ts:14]: This is one specific pool, we may run this agent for more pools too.

File: Curve-finance-suite/Remove-Imbalace-Liquidity/src/agent.ts [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+      ["uint256[3]","uint256[3]","uint256", "uint256"],
+      metadata!.data
+      );
+    
+    return Finding.fromObject({
+      name: "RemoveLiquidityImbalance event Detected",
+      description: 'RemoveLiquidityImbalance event emitted',
+      alertId: alertID,
+      severity: FindingSeverity.Low,
+      type: FindingType.Info,
```
Comment (mauricio1802) [inline Curve-finance-suite/Remove-Imbalace-Liquidity/src/agent.ts:30]: Maybe we can add `protocol: Curve Finance` here.

File: Curve-finance-suite/Remove-Imbalace-Liquidity/src/agent.ts (hunk 5 above)
Comment (mauricio1802) [reply Curve-finance-suite/Remove-Imbalace-Liquidity/src/agent.ts:14]: Let's use a list of contract addresses here ( from `https://curve.readthedocs.io/ref-addresses.html#base-pools` for example) or specify in the `README` that this is only checking `StableSwap3Pool`.

**Review by mauricio1802 — CHANGES_REQUESTED** [review]

**Review by mauricio1802 — APPROVED** [review]

**Pull request merged**